lib/whitehall_importer/integrity_checker.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class IntegrityChecker
+    attr_reader :edition
+
+    def initialize(edition)
+      @edition = edition
+    end
+
+    def valid?
+      problems.empty?
+    end
+
+    def problems
+      content_problems + image_problems + organisation_problems
+    end
+
+  private
+
+    def content_problems
+      problems = []
+
+      %w(base_path
+         title
+         description
+         document_type
+         schema_name).each do |attribute|
+        if publishing_api_content[attribute] != proposed_payload[attribute]
+          problems << "#{attribute} doesn't match"
+        end
+      end
+
+      problems << "body text doesn't match" unless body_text_matches?
+
+      problems
+    end
+
+    def body_text_matches?
+      proposed_body_text = proposed_payload.dig("details", "body")
+      publishing_api_body_text = publishing_api_content.dig("details", "body")
+
+      Sanitize.clean(publishing_api_body_text).squish == Sanitize.clean(proposed_body_text).squish
+    end
+
+    def image_problems
+      proposed_image_payload = proposed_payload.dig("details", "image") || {}
+      publishing_api_image = publishing_api_content.dig("details", "image") || {}
+
+      problems = []
+
+      %w(alt_text caption).each do |attribute|
```

Comment:
You can use an each_with_object or an inject to avoid the need to create and return the problems array.